src/js/view/controls/components/timeslider.ts

```diff
@@ -297,15 +297,15 @@ class TimeSlider extends Slider {
         addClass(timeTip.el, 'jw-open');
 
         const timeTipWidth = timeTip.getWidth();
-        const widthPct = railBounds.width / 100;
         const tolerance = playerWidth - railBounds.width;
-        let timeTipPct = 0;
+        let timeTipPixels = 0;
         if (timeTipWidth > tolerance) {
-            // timeTip may go outside the bounds of the player. Determine the % of tolerance needed
-            timeTipPct = (timeTipWidth - tolerance) / (2 * 100 * widthPct);
+            // timeTip may go outside the bounds of the player. Determine the amount of tolerance needed in pixels.
+            timeTipPixels = (timeTipWidth - tolerance) / 2;
         }
-        const safePct: number = parseFloat(Math.min(1 - timeTipPct, Math.max(timeTipPct, pct)).toFixed(3)) * 100;
-        transform(timeTip.el, `translateX(${safePct}%)`);
+        const safePixels: number = parseFloat(Math.min(railBounds.width - timeTipPixels,
+            Math.max(timeTipPixels, position)).toFixed(3));
+        transform(timeTip.el, `translateX(${safePixels}px)`);
```

Comment:
```suggestion
        const safePixels: number = parseInt(Math.min(railBounds.width - timeTipPixels,
            Math.max(timeTipPixels, position)));
        transform(timeTip.el, `translateX(${safePixels}px)`);
```

Not all browsers handle half pixels. May be better to drop the extra calculation all together. Up to you, approved!

Reply:
The flooring to thousands (`parseFloat` on `toFixed(3)`) was for percentages. For pixels we can and should allow sub pixel values but we should not floor we should round, and don't need to update CSS on every thousandth change. a quarter or half pixel change would be enough.

```suggestion
        const safePixels: number = Math.round(Math.min(railBounds.width - timeTipPixels,
            Math.max(timeTipPixels, position)) * 4) / 4;
        transform(timeTip.el, `translateX(${safePixels}px)`);
```